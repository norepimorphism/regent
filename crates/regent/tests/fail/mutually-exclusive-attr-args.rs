fn main() {
    // Mutually-exclusive paths, same semantics.
    #[regent::bitwise(size = 1, width = 8)]
    struct A;
    // Mutually-exclusive paths, different semantics.
    #[regent::bitwise(size = 1, width = 1)]
    struct B;
}
