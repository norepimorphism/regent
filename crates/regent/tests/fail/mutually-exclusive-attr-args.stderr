error: `size` and `width` arguments are mutually exclusive
 --> tests/fail/mutually-exclusive-attr-args.rs:3:5
  |
3 |     #[regent::bitwise(size = 1, width = 8)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `regent::bitwise` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `size` and `width` arguments are mutually exclusive
 --> tests/fail/mutually-exclusive-attr-args.rs:6:5
  |
6 |     #[regent::bitwise(size = 1, width = 1)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `regent::bitwise` (in Nightly builds, run with -Z macro-backtrace for more info)
